core-libs/setup/ssr/engine/cx-common-engine.spec.ts

```diff
@@ -0,0 +1,143 @@
+// eslint-disable-next-line import/no-unassigned-import
+import '@angular/compiler';
+
+import { Component, InjectionToken, NgModule, inject } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { ServerModule } from '@angular/platform-server';
+import { PROPAGATE_SERVER_ERROR_RESPONSE } from '../error-handling/server-error-response/propagate-server-error-response';
+import { CmsPageNotFoundServerErrorResponse } from '../public_api';
+import { CxCommonEngine } from './cx-common-engine';
+
+// Test how the cxCommonEngineWrapper handles successful server-side rendering
+@Component({ selector: 'cx-mock', template: 'some template' })
+export class SuccessComponent {}
+
+@NgModule({
+  imports: [BrowserModule, ServerModule],
+  declarations: [SuccessComponent],
+  bootstrap: [SuccessComponent],
+})
+export class SuccessServerModule {}
+
+// Test how the cxCommonEngineWrapper handles propagated server errors
+@Component({
+  selector: 'cx-response',
+  template: ``,
+})
+export class WithPropagatedErrorComponent {
+  constructor() {
+    inject(PROPAGATE_SERVER_ERROR_RESPONSE)(
+      new CmsPageNotFoundServerErrorResponse({ message: 'test error' })
+    );
+  }
+}
+
+@NgModule({
+  imports: [BrowserModule, ServerModule],
+  declarations: [WithPropagatedErrorComponent],
+  bootstrap: [WithPropagatedErrorComponent],
+})
+export class WithPropagatedErrorServerModule {}
+
+// Test that the cxCommonEngineWrapper doesn't override providers
+// If SOME_TOKEN not provided, test how the cxCommonEngineWrapper handles APP_INITIALIZER errors
```

Comment:
nitpick: this comment mentions `cxCommonEngineWrapper` which got removed from this PR

Note: there are other places in this file with leftover `cxCommonEngineWrapper` mentions

Let's fix it.